examples/webpack.config.local.js

```diff
@@ -131,32 +133,53 @@ const BABEL_RULE = {
   }
 };
 
-function addLocalDevSettings(config, exampleDir) {
+function addLocalDevSettings(sourceConfig, exampleDir) {
   const LOCAL_DEV_CONFIG = makeLocalDevConfig(exampleDir);
-  config = Object.assign({}, LOCAL_DEV_CONFIG, config);
+  const config = {...LOCAL_DEV_CONFIG, ...sourceConfig};
+
   config.resolve = config.resolve || {};
-  config.resolve.alias = config.resolve.alias || {};
-  Object.assign(config.resolve.alias, LOCAL_DEV_CONFIG.resolve.alias);
+  config.resolve = {
+    ...config.resolve,
+    alias: {
+      ...(config.resolve ? config.resolve.alias : null),
+      ...LOCAL_DEV_CONFIG.resolve.alias
+    }
+  };
+
+  config.module = {
+    ...config.module,
+    rules: [
+      ...(config.module ? config.module.rules : null),
+      ...LOCAL_DEV_CONFIG.module.rules
+    ]
+  };
+
+  config.plugins = [
+    ...config.plugins,
+    // Inject Current package version
+    new webpack.DefinePlugin({
```

Comment:
by injecting the plugin we make sure it doesn't get overridden by local webpack specific configuration

Reply:
you can use the same `search-and-replace` plugin here

Reply:
Done